web/models.py

```
@@ -213,6 +215,23 @@ def save(self, *args, **kwargs):
+            self.facts = []
+        for fact in self.facts:
+            if fact.get("text") == fact_text:
+                return False
+
+        self.facts.append({"text": fact_text, "generated_at": timezone.now().isoformat()})
+        self.save(update_fields=["facts"])
+        return True
+
+    def get_random_fact(self) -> Optional[str]:
+        if self.facts:
+            selected_fact = random.choice(self.facts)
```

Ruff (0.8.2)

Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)